src/main.ts

```diff
@@ -0,0 +1,15 @@
+export * from './terrain';
+export * from './analysis';
+export * from './basicTypes';
+export * from './brownian';
+export * from './core';
```

Comment:
Hmmm ... not sure about this one. Microsoft wrote a doc page that [discourages using modules and namespaces in the same file](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html). I tried similar stuff in the past and I always ran into some kind of absolute weirdness. It really seems as if Typescript does not like mixing up modules and namespaces. However, it was too long ago so that I do not remember what happened exactly.

Reply:
As far as I'm aware, the whole purpose of namespaces is to merge declarations like this. Nothing on the page you linked that I can see warns against combining them. Regardless, I don't care so much how it's accomplished, I just think `Terrain.Terrain` is awkward and it should be possible to construct an API signature that doesn't repeat itself 🙂